.circleci/config.yml

```diff
@@ -20,7 +20,135 @@ parameters:
+      - save_cache:
+          key: dependency-cache-{{ checksum "package-lock.json" }}
+          paths:
+            - node_modules
+            - ~/.cache
+      - run:
+          command: |
+            npm run build
+      - run: # Run linter after build because the integration test code depend on compiled typescript...
+          name: Linter check
+          command: npm run lint
+      - run:
+          name: Link shell scripts
+          command: node_modules/.bin/shellcheck ./script/**/*[^utils] ./script/utils/**
+      - persist_to_workspace:
+          root: .
+          paths:
+            - node_modules
+            - build
+            - dist
+            - .cache/Cypress
+
+  unit_test:
+    parallelism: 4
+    executor:
+      name: hmpps/node
+      tag: << pipeline.parameters.node-version >>
+    steps:
+      - checkout
+      - attach_workspace:
+          at: ~/app
+      - restore_cache:
+          key: dependency-cache-{{ checksum "package-lock.json" }}
+      - run:
+          name: unit tests
+          command: |
+            TESTS=$(circleci tests glob "server/**/*.test.ts" | circleci tests split --split-by=timings)
+            npm run test:ci $TESTS
+      - run:
+          name: collect coverage data
+          command: |
+            mv ./coverage/coverage-final.json ./coverage/coverage_${CIRCLE_NODE_INDEX}.json
+      - store_test_results:
+          path: test_results
+      - store_artifacts:
+          path: test_results/unit-test-reports.html
+      - persist_to_workspace:
+          root: .
+          paths:
+            - coverage
+
+  coverage:
+    executor:
+      name: hmpps/node
+      tag: << pipeline.parameters.node-version >>
+    steps:
+      - checkout
+      - attach_workspace:
+          at: ~/app
+      - run:
+          name: Check Coverage
+          command: |
+            npx nyc merge ./coverage/ ./coverage/.nyc_output
+            npx nyc report -t ./coverage --reporter=text --reporter=text-summary
+            npx nyc check-coverage -t ./coverage
+
+  integration_test:
+    parallelism: 4
+    executor:
+      name: integration-tests
+    steps:
+      - checkout
+      - attach_workspace:
+          at: ~/app
+      - run:
+          name: Install missing OS dependency
+          command: sudo apt-get install libxss1
+      - restore_cache:
+          key: dependency-cache-{{ checksum "package-lock.json" }}
+      - run:
+          name: Get wiremock
+          command: curl -o wiremock.jar https://repo1.maven.org/maven2/com/github/tomakehurst/wiremock-standalone/2.27.1/wiremock-standalone-2.27.1.jar
+      - run:
+          name: Run wiremock
+          command: java -jar wiremock.jar --port 9999
+          background: true
+      - run:
+          name: Run the node app.
+          command: npm run compile-sass && npm run start-feature
+          background: true
+      - run:
+          name: Wait for node app to start
+          command: sleep 5
```

Comment:
As a future optimisation could it would be good to know more precisely when the node app has started

Reply:
Yeah, I'm sure we did this on another project. It should just be a case of polling a port

Reply:
Yeah we did/do something similar in ap-tools

Reply:
Oh good call - I'll nick that

Reply:
That works!